hw/ip/lc_ctrl/data/lc_ctrl.hjson

```diff
@@ -501,21 +501,28 @@
               "excl:CsrAllTests:CsrExclCheck"],
       fields: [
         { bits: "0"
-          name: "READY"
+          name: "INITIALIZED"
```

Comment:
i can't fully remember...but i think there might be some top level enums that are somewhat related to this.
We'll probably want to change those.

Reply:
https://cs.opensource.google/opentitan/opentitan/+/master:hw/top_earlgrey/dv/env/chip_env_pkg.sv;l=95-104?q=lc_ctrl_status_e&ss=opentitan%2Fopentitan

there's probably a better way to do this..sorry my brain isn't really working right now.

Reply:
I changed the enums in DIFs and DV... was there another place where we had something like that?

@alphan does this affect ROM? The behavior of the READY bit has not changed, it is just at a different offset now.

Reply:
The code that polls the `READY` bit is [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom_start.S;drc=ff6f87515d0fac569fb32534b8fe78a329689ec4;l=276). It uses macros for the IP base address, register offset, and bit index. LGTM and can you also take a quick look to confirm?

Just to be clear: we will spin while `READY` is 0 and reset when it's 1 (if the CPU is still executing).

Reply:
@alphan i looked through the code a bit just to make sure i follow..
after waiting out the spin cycles, you basically poll the ready bit right? So if ready is '0', like you said, you will spin in place (allowing rma to happen).  If it is not, you will continue on with normal executing assuming someone else hasn't halted the processor?

Reply:
just to confirm one more thing and maybe a suggestion:

1. it might be good to move the jitter enable "after" the rma loop, just in case that affects rma in any way.
2. does the SPIN_CYCLES refer to the absolute number of cycles up to that point? Ie, SPIN_CYCLES would also need to cover the amount of time required before the rma bootstrap stuff?

Reply:
> Just to be clear: we will spin while `READY` is 0 and reset when it's 1 (if the CPU is still executing).
this is OK. if READY is 0, a life cycle transition has been initiated.

Reply:
> just to confirm one more thing and maybe a suggestion:
> 
> 1. it might be good to move the jitter enable "after" the rma loop, just in case that affects rma in any way.
> 2. does the SPIN_CYCLES refer to the absolute number of cycles up to that point? Ie, SPIN_CYCLES would also need to cover the amount of time required before the rma bootstrap stuff?

1. I agree, this would be good to do.
2. this appears to be using `mcycle` to compare against, meaning that it's the absolute number of CPU clock cycles since it has been reset. not sure it matters, but this limits us to 2^32/100MHz = ~42 seconds if only using the lower 32bit of `mcycle`.

Reply:
> @alphan i looked through the code a bit just to make sure i follow.. after waiting out the spin cycles, you basically poll the ready bit right? So if ready is '0', like you said, you will spin in place (allowing rma to happen). If it is not, you will continue on with normal executing assuming someone else hasn't halted the processor?

Correct.

Reply:
> just to confirm one more thing and maybe a suggestion:
> 
> 1. it might be good to move the jitter enable "after" the rma loop, just in case that affects rma in any way.

jitter_enable depends on AST init and IIRC you preferred performing AST init before RMA busy loop. I can move things around but maybe we should chat offline.

> 2. does the SPIN_CYCLES refer to the absolute number of cycles up to that point? Ie, SPIN_CYCLES would also need to cover the amount of time required before the rma bootstrap stuff?

SPIN_CYCLES refer to the absolute number of cycles that we will wait for after resetting `mcycle` [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom_start.S;drc=ff6f87515d0fac569fb32534b8fe78a329689ec4;l=221). Next, we check and busy loop while the `READY` bit is `0`. So, SPIN_CYCLES is the number of cycles that we will wait until RMA can start. We will then busy loop as long as `READY` is `0`.

Reply:
>  not sure it matters, but this limits us to 2^32/100MHz = ~42 seconds if only using the lower 32bit of `mcycle`.

Yes, but SPIN_CYCLES is like the bridge until RMA transition starts. I thought 32-bits is more than enough since after that we busy loop while `READY` is 0. Please let me know if I missed something.

Reply:
Ok, good, just wanted to make sure this was intentional. I think it should be fine, since connecting to the JTAG an issuing the transaction should be way faster than that.

Reply:
@alphan sorry i should have clarified. If possible, it would be nice to do 
@zi-v for confirmation.
1. ast init
2. rma bootstrap
3. jittery enable. 

It's not deal breaker if we do not though.

Reply:
Thanks for clarifying, I need to massage that code a little bit because jittery was embedded in the ast init check but I can pull it out if you think it's worth it.

Reply:
definitely not critical IMO. 
but lets see if @zi-v disagrees.

Reply:
I think it is OK to enable jitter before RMA branch as long as we check this particular scenario in DV.

Reply:
SGTM